site/docs/v1/tech/guides/auth0-migration.adoc

```diff
@@ -0,0 +1,380 @@
+
+[source,json,title=Sample Auth0 password hash export download]
+----
+{"_id":{"$oid":"60425dc43519d90068f82973"},"email_verified":false,"email":"test2@example.com","passwordHash":"$2b$10$Z6hUTEEeoJXN5/AmSm/4.eZ75RYgFVriQM9LPhNEC7kbAbS/VAaJ2","password_set_date":{"$date":"2021-03-05T16:35:16.775Z"},"tenant":"dev-rwsbs6ym","connection":"Username-Password-Authentication","_tmp_is_unique":true}
+{"_id":{"$oid":"60425da93519d90068f82966"},"email_verified":false,"email":"test@example.com","passwordHash":"$2b$10$CSZ2JarG4XYbGa.JkfpqnO2wrlbfp5eb5LScHSGo9XGeZ.a.Ic54S","password_set_date":{"$date":"2021-03-05T16:34:49.502Z"},"tenant":"dev-rwsbs6ym","connection":"Username-Password-Authentication","_tmp_is_unique":true}
+----
+
+Now that you have both the user data and the passwords hashes, you can import your users.
+
+== Importing Users
+
+There are a few steps to take to import users. 
+
+* Prepare your FusionAuth installation
+* Get the import script 
+* Install needed gems
+* Customize the script
+* Run the script
+* Test
+
+=== Set up FusionAuth
+
+You need to set up FusionAuth so migrated user data can be stored. As mentioned above, this guide assumes you have FusionAuth installed. If you don't, link:/docs/v1/tech/installation-guide/[view our installation guides] and get it installed before proceeding further.
+
+==== Create a Test Tenant
+
+It is best to create a separate tenant for a migration testing. Tenants logically isolate configuration settings and users. If a migration goes awry or you need to redo it after tweaking the script, you can delete the test tenant and start with a clean system. To add a tenant, navigate to [breadcrumb]#Tenants# and choose the green plus sign.
+
+image::guides/auth0-migration/list-of-tenants-add-highlighted.png[Adding a tenant.,width=1200,role=bottom-cropped]
+
+Give it a descriptive name like `Auth0 import test` and save it. You shouldn't need to modify any of the other configuration options to test importing users. Save the tenant.
+
+image::guides/auth0-migration/add-tenant.png[The tenant creation screen.,width=1200,role=bottom-cropped]
+
+Record the Id of the tenant, which will be something like `25c9d123-8a79-4edd-9f76-8dd9c806b0f3`. You'll use this in the import script.
+
+image::guides/auth0-migration/list-of-tenants.png[The tenant list.,width=1200,role=bottom-cropped]
+
+==== Create a Test Application
+
+Applications are anything that a user can log in to. In FusionAuth there's no differentiation between web applications, SaaS applications, APIs and native apps. To add an application, navigate to [breadcrumb]#Applications# and choose the green plus sign. Give the application a descriptive name like `Auth0 application`.
+
+Navigate to the [breadcrumb]#OAuth# tab and add an entry to [field]#Authorized redirect URLs#. Use a value such as `\https://fusionauth.io` for testing. You shouldn't need to modify any of the other configuration options to test importing users. Save the application. 
```

Comment:
Is this clear that this URL is just a test value, and it should eventually be your callback? If someone is migration from Auth0 and is already using OAuth/OIDC, they may know their actual redirect URL?

Reply:
I mention that further down in the 'testing' section. I wanted to avoid the entire authorization code flow discussion, but will refine the testing url text.